agent/php_user_instrument.c

```diff
@@ -309,6 +321,47 @@ static void nr_php_add_custom_tracer_common(nruserfn_t* wraprec) {
   nr_wrapped_user_functions = wraprec;
 }
 
+bool nr_php_wraprec_matches(nruserfn_t* p, zend_function* func) {
+#if ZEND_MODULE_API_NO < ZEND_7_0_X_API_NO
+  /*
+   * Not compatible with PHP less than 7.
+   */
+  (void)func;
+  (void)p;
+  return false;
+#else
+  char* klass = NULL;
+
+  /*
+   * Optimize out string manipulations; don't do them if you don't have to.
+   * For instance, if funcname doesn't match, no use comparing the classname.
+   */
+
+  if (NULL == p) {
+    return false;
+  }
+  if ((NULL == func) || (ZEND_USER_FUNCTION != func->type)) {
+    return false;
+  }
+  if (NULL == func->common.function_name) {
+    return false;
+  }
+
+  if (0 != nr_stricmp(p->funcnameLC, ZSTR_VAL(func->common.function_name))) {
+    return false;
+  }
+  if (NULL == func->common.scope || NULL == func->common.scope->name) {
+    return false;
+  }
+  klass = ZSTR_VAL(func->common.scope->name);
```

Comment:
fix: apparently `NULL` and `NULL` are valid comparisons in nr_strcmp
```suggestion
  if (NULL != func->common.scope && NULL != func->common.scope->name) {
    klass = ZSTR_VAL(func->common.scope->name);
  }
```